codec_fixed.go

```diff
@@ -20,6 +21,12 @@ func createDecoderOfFixed(schema Schema, typ reflect2.Type) ValDecoder {
 		return &fixedCodec{arrayType: typ.(*reflect2.UnsafeArrayType)}
 
 	case reflect.Struct:
+		if typ.String() == "avro.LogicalDuration" {
+			if fixed.Size() != 12 {
+				return &errorDecoder{err: fmt.Errorf("avro: duration expects size=12, got size=%d", fixed.Size())}
+			}
+			return &fixedDurationCodec{}
+		}
 		ls := fixed.Logical()
 		if typ.RType() != ratRType || ls == nil || ls.Type() != Decimal {
```

Comment:
There are some issues here:
1. It is best not to check against type strings but rather against rtypes like `ratRType` below.
2. You are not checking the logical type

The struct block should actually look like this now:
```
ls := fixed.Logical()
		if ls == nil {
			break
		}
		switch {
		case typ.RType() == durRType && ls.Type() == Duration:
			return &fixedDurationCodec{}
		case typ.RType() == ratRType && ls.Type() == Decimal:
			dec := ls.(*DecimalLogicalSchema)
			return &fixedDecimalCodec{prec: dec.Precision(), scale: dec.Scale(), size: fixed.Size()}
		}
```